sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_do_while.py

```diff
@@ -54,4 +54,23 @@ def pipeline_with_do_while_node():
         _from_component_func=True,
     )
     do_while_node.set_limits(max_iteration_count=max_iteration_count)
+
+    def _infer_and_update_body_input_from_mapping():
+        # pylint: disable=protected-access
+        for output_name, body_input in mapping.items():
+            # if input type is specified, no need to infer and skip
+            if body_input.type is not None:
+                continue
+            inferred_type = body.outputs[output_name].type
+            # update node input
+            body_input._meta._is_inferred_optional = True
```

Comment:
why do we need this bool?

Reply:
If not specify here, the optional will be None, then the parameter will be handled as required. So update this here.

Reply:
This bool is also used when resolve input, for such optional input without path, will bypass resolvement.